postgis_helpers.go

```diff
@@ -0,0 +1,111 @@
+package tilenol
+
+import (
+	"bytes"
+	"database/sql"
+	"time"
+
+	"github.com/paulmach/orb"
+	"github.com/paulmach/orb/encoding/wkb"
+)
+
+// RowsToMaps converts Go's sql.Rows data structure into a list of maps where the key is a string
+// column name, and the value is the raw SQL value
+func RowsToMaps(rows *sql.Rows) ([]map[string]interface{}, error) {
+	var maps []map[string]interface{}
+
+	cols, err := rows.Columns()
+	if err != nil {
+		return maps, err
+	}
+
+	for rows.Next() {
+		row := make([]interface{}, len(cols))
+		for idx := range cols {
+			row[idx] = new(RowScanner)
+		}
+		err := rows.Scan(row...)
+		if err != nil {
+			return maps, err
+		}
+		m := make(map[string]interface{})
+		for idx, col := range cols {
+			var s = row[idx].(*RowScanner)
+			if s.valid {
+				m[col] = s.value
+			}
+		}
+		maps = append(maps, m)
+	}
+
+	return maps, nil
+}
+
+// Loosely adapted "Scanner" implementation that knows how to unmarshal data from the raw database
+// response
+type RowScanner struct {
+	valid bool
+	value interface{}
+}
+
+func (s *RowScanner) getBytes(src interface{}) []byte {
+	if a, ok := src.([]uint8); ok {
```

Comment:
If I grok `getsBytes` correctly, what it does is "If `src` is actually something you can get an array of bytes from, then return it, else return `nil`.

What I am not groking is the `ok` part. -- This looks like a self-referencing variable to me, without an actual value ever being assigned to it? -- That's probably some nifty Go idiom that I don't know?

Reply:
Yeah, that's basically it :+1:. The `ok` part is Go's way of doing a checked "type assertion". Basically when you say:

```go
intVal, isItActuallyAnInt := someVal.(int64)
```

`intVal` will contain the casted value of `someVal`, and `isItActuallyAnInt` will contain a boolean value indicating whether or not the value could actually be casted to an `int64`. Note that if we use the single-value return version of a type assertion:

```go
intVal := someVal.(int64)
```

Then if we can't assert the type as an `int64`, this will cause a panic at runtime.